collector-sqs/src/main/java/zipkin/collector/awssqs/AwsSqsSpanProcessor.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2016 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.awssqs;
+
+import com.amazonaws.handlers.AsyncHandler;
+import com.amazonaws.services.sqs.AmazonSQSAsync;
+import com.amazonaws.services.sqs.model.DeleteMessageBatchRequestEntry;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.MessageAttributeValue;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import zipkin.Codec;
+import zipkin.Component;
+import zipkin.Span;
+import zipkin.collector.Collector;
+import zipkin.internal.Nullable;
+import zipkin.storage.Callback;
+
+
+final class AwsSqsSpanProcessor implements Closeable, Component {
+
+  private static final Logger log = LoggerFactory.getLogger(AwsSqsSpanProcessor.class);
+
+  private final AmazonSQSAsync client;
+  private final Collector collector;
+  private final String queueUrl;
+  private final int waitTimeSeconds;
+  private final AtomicBoolean closed = new AtomicBoolean(false);
+  private final AtomicReference<CheckResult> status = new AtomicReference<>(CheckResult.OK);
+
+  AwsSqsSpanProcessor(AmazonSQSAsync client, Collector collector, String queueUrl,
+      int waitTimeSeconds) {
+    this.client = client;
+    this.collector = collector;
+    this.queueUrl = queueUrl;
+    this.waitTimeSeconds = waitTimeSeconds;
+  }
+
+  @Override public CheckResult check() {
+    return status.get();
+  }
+
+  @Override public void close() throws IOException {
+    if (!closed.getAndSet(true)) {
+      client.shutdown();
+    }
+  }
+
+  AwsSqsSpanProcessor run() {
+    final ReceiveMessageRequest request = new ReceiveMessageRequest(queueUrl)
```

Comment:
style: please only add final on parameters or local variables where we should pay attention to that.